__tests__/TextPath.test.js

```diff
@@ -109,22 +126,24 @@ describe('<TextPath />', () => {
     describe('props change', () => {
         it('should call setText() with null to clear the existing text', () => {
             const wrapper = mount(
-                <Map>
+                <MapContainer>
                     <TextPath text={mockText} />
-                </Map>
+                </MapContainer>
             );
             updateProps(wrapper, { orientation: 'flip' });
-            expect(setTextSpy).toHaveBeenNthCalledWith(2, null);
+            // TODO: There appears to be a bug where react-leaflet is recreating the element
```

Comment:
This is an interesting one!  I can see it indeed calling the create method after props update when trying this library in a barebones v3 project.

This does slightly worry me as it looks to be unnecessarily executing and creating new instances of `LeafletPolyline` on every prop change.  I did a very quick test by updating the TextPath props quite a few times and after profiling in the dev tools I could see the memory going up quite a lot, significantly more than if the TextPath was commented out.

It feels like this may be a bug in the react-leaflet, highly possible with v3 being quite new.  Do you think you'd be able to investigate it a bit more, even create a minimal reproduction to raise with Paul from react-leaflet?

I'd try and help but unfortunately I'm pretty maxed out at work so don't have any time to spare!  Completely understand if it's the same with you of course too.

Reply:
Thanks for trying to figure it out.

I managed to whittle it down to a reproducable CodeSandbox and submitted the issue here:

PaulLeCam/react-leaflet#830

Reply:
Superb, thank you!